metadata-ingestion/src/datahub/configuration/common.py

```diff
@@ -243,15 +243,21 @@ def allow_all(cls) -> "AllowDenyPattern":
         return AllowDenyPattern()
 
     def allowed(self, string: str) -> bool:
-        for deny_pattern in self.deny:
-            if re.match(deny_pattern, string, self.regex_flags):
-                return False
+        if self.denied(string):
+            return False
 
         return any(
             re.match(allow_pattern, string, self.regex_flags)
             for allow_pattern in self.allow
         )
 
+    def denied(self, string: str) -> bool:
```

Comment:
we probably don't want this to be part of the "public" interface

```suggestion
    def _denied(self, string: str) -> bool:
```